pkg/azure/utils.go

```diff
@@ -601,21 +601,10 @@ func (d *MachinePlugin) createVMNicDisk(req *driver.CreateMachineRequest) (*comp
 // deleteVMNicDisks deletes the VM and associated Disks and NIC
 func (d *MachinePlugin) deleteVMNicDisks(ctx context.Context, clients spi.AzureDriverClientsInterface, resourceGroupName string, VMName string, nicName string, diskName string, dataDiskNames []string) error {
 
-	// We try to fetch the VM, detach its data disks and finally delete it
-	if vm, vmErr := clients.GetVM().Get(ctx, resourceGroupName, VMName, ""); vmErr == nil {
-
-		if detachmentErr := waitForDataDiskDetachment(ctx, clients, resourceGroupName, vm); detachmentErr != nil {
-			return detachmentErr
-		}
-		if deleteErr := DeleteVM(ctx, clients, resourceGroupName, VMName); deleteErr != nil {
-			return deleteErr
-		}
-
-		OnARMAPISuccess(prometheusServiceVM, "VM Get was successful for %s", *vm.Name)
-	} else if !NotFound(vmErr) {
-		// If some other error occurred, which is not 404 Not Found (the VM doesn't exist) then bubble up
-		return OnARMAPIErrorFail(prometheusServiceVM, vmErr, "vm.Get")
+	if deleteErr := DeleteVM(ctx, clients, resourceGroupName, VMName); deleteErr != nil && !NotFound(deleteErr) {
+		return deleteErr
```

Comment:
if we don't wait for disk detachment (which could take 10 min) 
then we'll directly move to the disk deletion below while disk are detaching
this could make things complicated as azure can't be trusted.

Reply:
sorry after taking a re-look , I realize that we are not even issuing a detach call now, and going for a direct delete. 
this solution would be deterministic only if `DeleteVM` returns after the detach and not before, otherwise we will go for disk deletion and it might fail and lead to inconsistencies.